lib/route/link/bridge.c

```diff
@@ -774,11 +917,38 @@ int rtnl_link_bridge_set_self(struct rtnl_link *link)
 	IS_BRIDGE_LINK_ASSERT(link);
 
 	bd->b_self |= 1;
+	bd->b_master = 0;
+	bd->ce_mask &= ~BRIDGE_ATTR_MASTER;
 	bd->ce_mask |= BRIDGE_ATTR_SELF;
 
 	return 0;
 }
 
+/**
+ * Set link change type to master
+ * @arg link		Link Object of type bridge
+ *
+ * This will set the bridge change flag to master, meaning that changes to
+ * be applied with this link object will be applied directly to the virtual
+ * device in a bridge instead of the physical device.
+ *
+ * @return 0 on success or negative error code
+ * @return -NLE_OPNOTSUP Link is not a bridge
+ */
+int rtnl_link_bridge_set_master(struct rtnl_link *link)
+{
+	struct bridge_data *bd = bridge_data(link);
+
+	IS_BRIDGE_LINK_ASSERT(link);
+
+	bd->b_master |= 1;
+	bd->b_self = 0;
```

Comment:
if `s_master` and `s_self` are only set to 1, why the `|= 1`?

Anyway, can we make those two fields `bool` and use `true`/`false` ?


Also, `b_master` is only an internal field, and it's always `!b_self`. Are there future plans for this? Just drop the redundant field?

Maybe it would be better to support instead the `IFLA_BRIDGE_FLAGS`, and not trying to abstract individual flags (`BRIDGE_FLAGS_SELF`, `BRIDGE_FLAGS_MASTER`).